### PR TITLE
feat: add classes to parse executionRequest in Flight RPC

### PR DESCRIPTION
**Pull request by no23reason**
Allow the users to parse the incoming executionRequest when invoking the Flight RPC function. Use the gooddata-sdk objects so that they can be easily reused in the gooddata-sdk itself.

Also, add the gooddata-flight-server to the CI pipelines, make sure the coverage is published and fix all warnings in the Dockerfile.

Fix some code smells along the way, see the last commit for more details.

JIRA: CQ-577
risk: low

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,8 +27,8 @@ RUN set -x \
   && true
 
 # install tox
-ENV PYTHON_TOX_VERSION 4.14.0
-ENV PYTHON_TOX_UV_VERSION 1.5.1
+ENV PYTHON_TOX_VERSION=4.14.0
+ENV PYTHON_TOX_UV_VERSION=1.5.1
```
Comment (hkad98) [inline Dockerfile:31]: Pls, bump tox to 4.14.1 and tox uv to 1.7.0. So it is in sync with tox-requirements.txt

File: gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# (C) 2024 GoodData Corporation
+from dataclasses import dataclass
+from typing import Optional
+
+from gooddata_sdk import Attribute, ComputeToSdkConverter, Filter, Metric
+
+
+@dataclass
```
Comment (hkad98) [inline gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py:8]: Consider using attrs instead.

File: gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        :param d: the dictionary to parse
+        """
+        return ExecutionRequest(
+            attributes=[ComputeToSdkConverter.convert_attribute(a) for a in d.get("attributes", [])],
+            metrics=[ComputeToSdkConverter.convert_metric(m) for m in d.get("measures", [])],
+            filters=[ComputeToSdkConverter.convert_filter(f) for f in d.get("filters", [])],
+        )
+
+
+@dataclass
```
Comment (hkad98) [inline gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py:42]: Consider using attrs instead.

File: Dockerfile (hunk 1 above)
Comment (no23reason) [reply Dockerfile]: updated

File: gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py (hunk 2 above)
Comment (no23reason) [reply gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py:8]: Let's not add this dependency until we need some of its features

File: gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py (hunk 3 above)
Comment (no23reason) [reply gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py:42]: Let's not add this dependency until we need some of its features

**Review by hkad98 — APPROVED** [review]

**Pull request merged**